packages/askui-nodejs/src/execution/execution-runtime.ts

```diff
@@ -84,18 +94,25 @@ export class ExecutionRuntime {
     return command;
   }
 
+  private async getImageIfRequired(instruction: string): Promise<string | undefined> {
+    const isImageRequired = await this.inferenceClient.isImageRequired(instruction);
+    if (!isImageRequired) return undefined;
```

Comment:
Does not return a promise of undefined.  --> Suprise

if condtion should contain always brackets for the body.

Reply:
My private coding style came through where I leave out the curly braces :D Going to add a linting rule so that never happens again ;P

But I don't get your first comment. Could you explain further?

Reply:
My fault. 

I thought when you are returning undefined in the function, that the reference of the promise is undefined instead of the promis is resolving with an undefined value. 

But this is a async function. So it wraps the return value in an promise